prov/rxd/src/rxd_atomic.c

```diff
@@ -245,8 +257,8 @@ static ssize_t rxd_atomic_inject(struct fid_ep *ep_fid, const void *buf,
 			0, NULL, 0, datatype, op);
 	if (!tx_entry)
 		goto out;
-
-	if (rxd_ep->peers[rxd_addr].peer_addr == FI_ADDR_UNSPEC)
+    peer_entry = ofi_bufpool_get_ibuf(rxd_ep->peer_pool.pool, rxd_addr);
+	if (peer_entry->peer_addr == FI_ADDR_UNSPEC)
```

Comment:
In some cases when we get  the peer entry, we also use or manipulate other fields of the peer_entry struct.  (for ex : rxd_cq.c  rxd_ep_recv_data we use rx_seq_no and rx_window). should we then write inline getters and setters for every field that is being used or written to of the peer_entry? or should we do it only for peer_addr, because there are a few functions which only use the peer_addr.